### ISSUE TITLE
intermittent twisted.test.test_pb.NSPTestCase.test_NSP failure on OS X

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2947]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#2947|
|Type|defect|
|Created|2007-12-11 12:45:56Z|
|Branch|https://github.com/twisted/twisted/tree/pb-test-cleanup-2947|

```
[ERROR]: twisted.test.test_pb.NSPTestCase.test_NSP

Traceback (most recent call last):
Failure: twisted.trial.util.DirtyReactorAggregateError: Reactor was unclean.
Selectables:
<Broker #0 on 54277>
```

<details><summary>Searchable metadata</summary>

```
trac-id__2947 2947
type__defect defect
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_pb_test_cleanup_2947 branches/pb-test-cleanup-2947
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1197377156000000 1197377156000000
changetime__1221241176000000 1221241176000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [24794]) Branching to 'pb-test-cleanup-2947'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun removed owner|
|-|-|

Should be fixed

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve set owner to @exarkun|
|-|-|

Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set status to `closed`|
|-|-|

(In [24807]) Merge pb-test-cleanup-2947

Author: exarkun
Reviewer: therve
Fixes: [#2947](https://github.com/twisted/twisted/issues/2947)

Fix `test_NSP` so that it waits for the server to disconnect after
the test completes so that there are no selectables left for
trial to report as making the reactor dirty.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
